Spotbugs error with Introspection and javax validation

### Steps to Reproduce

(also explained in git-repo: https://github.com/Kidlike/mn-spotbugs-sample)

- Start from a clean project generated by micronaut-cli version 3.0.3 with `mn create-app --build=maven mn-spotbugs-sample`
- Add a `@NotBlank` validation and spotbugs maven plugin
- Run `mvn clean install`

### Expected Behaviour

The maven build should not fail; more specifically the spotbugs plugin that detects a `NP_NONNULL_PARAM_VIOLATION` inside Micronaut code.

### Actual Behaviour

- The maven build fails with
   ```
   [ERROR] High: Null passed for non-null parameter of io.micronaut.core.beans.AbstractBeanIntrospection.indexProperty(Class, String, String) in new mn.spotbugs.sample.$SampleDto$Introspection() [mn.spotbugs.sample.$SampleDto$Introspection] In <Unknown> NP_NONNULL_PARAM_VIOLATION
   ```
- The detected `NP_NONNULL_PARAM_VIOLATION` is inside Micronaut code.
   More specifically, a micronaut-generated class (`...$Introspection`), is calling a micronaut-core method ([AbstractBeanIntrospection#indexProperty](https://github.com/micronaut-projects/micronaut-core/blob/3.0.x/core/src/main/java/io/micronaut/core/beans/AbstractBeanIntrospection.java#L282)) with `null` on a parameter that is marked as `@NonNull`.

### Environment Information

- **Operating System**: Fedora 33, Kernel `5.13.19-100.fc33.x86_64`, KDE Plasma 5.20.5
- **Micronaut Version:** 3.0.3
- **JDK Version:** AdoptOpenJDK 11.0.8+10 (2020-07-14)

### Example Application

- https://github.com/Kidlike/mn-spotbugs-sample

--- a/core/src/main/java/io/micronaut/core/beans/AbstractBeanIntrospection.java
+++ b/core/src/main/java/io/micronaut/core/beans/AbstractBeanIntrospection.java
@@ -287,7 +287,7 @@ public abstract class AbstractBeanIntrospection<T> implements BeanIntrospection<
     protected final void indexProperty(
             @NonNull Class<? extends Annotation> annotationType,
             @NonNull String propertyName,
-            @NonNull String annotationValue) {
+            @Nullable String annotationValue) {
         indexProperty(annotationType, propertyName);
         if (StringUtils.isNotEmpty(annotationValue) && StringUtils.isNotEmpty(propertyName)) {
             if (indexedValues == null) {

